agent/consul/state/catalog_test.go

```diff
@@ -1455,7 +1455,8 @@ func TestStateStore_EnsureService(t *testing.T) {
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
-	if idx != 30 {
```

Comment:
this changes is not necessary

Reply:
The test fails without this change, which follows the intended change in behavior. Could you elaborate why you think it's not necessary?